docs/content/preview/migrate/known-issues.md

```diff
@@ -253,6 +208,357 @@ CREATE TABLE address (
+
+**Description**: If you have a temporary table defined in a function or stored procedure in MySQL and you have a `drop temporary table` statement associated with it, the schema gets exported as is, which is an invalid syntax in YugabyteDB.
+
+**Workaround**: Manually remove the temporary clause from the drop statement.
+
+**Example**
+
+An example schema on the source database is as follows:
+
+```sql
+/* procedure definition */
+delimiter //
+CREATE PROCEDURE foo(p_id int)
+deterministic
+  BEGIN
+    DROP TEMPORARY TABLE IF EXISTS temp;
+    CREATE TEMPORARY TABLE temp(id int, name text);
+    INSERT INTO temp(id,name) select id,p_name from bar where p_id=id;
+    SELECT name FROM temp;
+  END//
+delimiter;
+```
+
+The exported schema is as follows:
+
+```sql
+CREATE OR REPLACE PROCEDURE foo (p_id integer) AS $body$
+  BEGIN
+    DROP TEMPORARY TABLE IF EXISTS temp;
+    CREATE TEMPORARY TABLE temp(id int, name text);
+    INSERT INTO temp(id,name) select id,p_name from bar where p_id=id;
+    SELECT name FROM temp;
+  END;
+$body$
+LANGUAGE PLPGSQL
+SECURITY DEFINER
+;
+```
+
+Suggested change to the schema is to remove the temporary clause from the drop statement as follows:
+
+```sql
+CREATE OR REPLACE PROCEDURE foo (p_id integer) AS $body$
+  BEGIN
+    DROP TABLE IF EXISTS temp;
+    CREATE TEMPORARY TABLE temp(id int, name text);
+    INSERT INTO temp(id,name) SELECT id,p_name FROM bar WHERE p_id=id;
+    SELECT name FROM temp;
+  END;
+$body$
+LANGUAGE PLPGSQL
+SECURITY DEFINER
+;
+```
+
+---
+
+#### Key defined for a table in functions/procedures cause issues
+
+**GitHub link**: [Issue #707](https://github.com/yugabyte/yb-voyager/issues/707)
+
+**Description**: If you have a basic _key_ defined for a table in a function/procedure, the schema is exported as is, and causes issues because using a key in YugabyteDB is an invalid syntax.
+
+**Workaround**: Manually remove the key from the exported schema, or create an index.
+
+**Example**
+
+An example schema on the source database is as follows:
+
+```sql
+/* function definition */
+
+delimiter //
+CREATE FUNCTION foo (p_id int)
+RETURNS varchar(20)
+READS SQL DATA
+  BEGIN
+    CREATE TEMPORARY TABLE temp(id int, name text,key(id));
+    INSERT INTO temp(id,name) SELECT id,p_name FROM bar WHERE p_id=id;
+    RETURN (SELECT name FROM temp);
+  END//
+```
+
+The exported schema is as follows:
+
+```sql
+CREATE OR REPLACE FUNCTION foo (p_id integer) RETURNS varchar AS $body$
+  BEGIN
+    CREATE TEMPORARY TABLE temp(id int, name text,key(id));
+    INSERT INTO temp(id,name) SELECT id,p_name FROM bar WHERE p_id=id;
+    RETURN (SELECT name FROM temp);
+  END;
+$body$
+LANGUAGE PLPGSQL
+SECURITY DEFINER
+;
+```
+
+Choose one from the following suggested changes to the schema.
+
+- Remove the key from the schema as follows:
+
+    ```sql
+    CREATE OR REPLACE FUNCTION foo (p_id integer) RETURNS varchar AS $body$
+      BEGIN
+        CREATE TEMPORARY TABLE temp(id int, name text,key(id));
```

Comment:
Please remove the `key` from this statement.

Reply:
good catch. Thanks!